lightning/src/ln/channelmanager.rs

```diff
@@ -7514,6 +7612,63 @@ This indicates a bug inside LDK. Please report this error at https://github.com/
 						}
 					});
 			},
+			HTLCSource::TrampolineForward { previous_hop_data, .. } => {
+				for current_previous_hop_data in previous_hop_data {
+					let prev_channel_id = current_previous_hop_data.channel_id;
```

Comment:
Am I correct that this handling is identical to `PreviousHopData`? Could be DRY-ed up into a method that we just call multiple times for trampoline hops. 

Also echo the naming suggestion [here](https://github.com/lightningdevkit/rust-lightning/pull/3711/files/a258d01a78c5a978727c7761f9188ba9458bc472..2e49bc3382b35f4deaf198ddcbaebb5cb61543ad#r2047619576)